src/customlabels.c

```diff
@@ -51,16 +51,48 @@ static bool eq(custom_labels_string_t l, custom_labels_string_t r) {
 
 #include <stdio.h>
 
-void custom_labels_labelset_print_debug(custom_labels_labelset_t *ls) {
-  unsigned ct = ls->count;
-  fprintf(stderr, "{");
-  for (unsigned i = 0; i < ct; ++i) {
-    custom_labels_label_t *lbl = &ls->storage[i];
-    fprintf(stderr, "%.*s: %.*s", (int)lbl->key.len, lbl->key.buf, (int)lbl->value.len, lbl->value.buf);
-    if (i != ct - 1)
-            fprintf(stderr, ", ");
-  }
-  fprintf(stderr, "}");
+char *custom_labels_labelset_debug_string(const custom_labels_labelset_t *ls) {
```

Comment:
As discussed on slack: use the internal string type so this doesn't break if someone tries to encode null bytes.

Reply:
Done now.